File: naTanjir/naTanjir.Model/Narudzba.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace naTanjir.Model
{
    public class Narudzba
    {
        public int NarudzbaId { get; set; }

        public int BrojNarudzbe { get; set; }

        public decimal UkupnaCijena { get; set; }

        public DateTime DatumKreiranja { get; set; }
        public int KorisnikId { get; set; }
        public string? StateMachine { get; set; }
        public virtual Korisnici Korisnik { get; set; }
        public int DostavljacId {  get; set; }
        public virtual ICollection<StavkeNarudzbe> StavkeNarudzbes { get; set; } = new List<StavkeNarudzbe>();



    }
}

File: naTanjir/UI/natanjir_mobile/lib/models/narudzba.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:natanjir_mobile/models/korisnici.dart';
import 'package:natanjir_mobile/models/stavke_narudzbe.dart';
part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  int? narudzbaId;
  int? brojNarudzbe;
  double? ukupnaCijena;
  String? datumKreiranja;
  String? stateMachine;
  int? korisnikId;
  Korisnici? korisnik;
  int? dostavljacId;
  List<StavkeNarudzbe>? stavkeNarudzbe;

  Narudzba();

  factory Narudzba.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
